### DISCUSSION TITLE
maya-usd in Maya 2019

### DISCUSSION THREAD

**Discussion by mreinhardmikros** [Autodesk/maya-usd#938] [Q&A]
Hello,
I'd like to ask some help/experience with building maya-usd for Maya 2019.
Right now I am able to load the plugin in 2019 and export a single geometry, but I am facing a hard crash when using the "Create->USD->Stage From File" command. (The exported usd file itself is fine, any file provokes a crash on import).
There's no Stack Trace whatsoever.
I have compiled USD from the 20.11 tag on the repo, using only the versions given in the OpenSource components help page for Maya 2019 (https://help.autodesk.com/view/MAYAUL/2019/ENU/?guid=__developer_Maya_SDK_MERGED_2019_Open_Source_Components_html), i.e: 
PYTHON 2.7.11
CLEW 0.10
BOOST 1.61.0
TBB 4.4.6
OPENEXR 2.2.1
ALEMBIC 1.7.5
OPENSUBDIV 3.3.3
PYOPENGL 3.1.5
PYSIDE2 5.11.0
JINJA2 2.0
OIIO 1.8.9
LIBTIFF 4-0-8
LIBPNG 1.6.34
EMBREE 3.2.2
OPENCOLORIO 1.0.9
MATERIALX  1.37.1
MARKUPSAFE 1.1.1

The **only** exception is GLEW:
I used GLEW 1.11.0 because USD requires symbols from OpenGL 4.5 which is not supported by the version specified by Maya (1.10.0)

Is there anybody who can tell me how I should deal with that?
Has anybody succeeded in running the plugin in 2019 ?
Thanks !
Marcel

**Comment by suesauer** [comment]
Hey Marcel, I started building in Maya 2019, got it working last year October. Was running 19.07 or something like that back then. Could be that Maya 1.10.0 doesn't work with newer USD pxr...

**Reply by suesauer** [reply]
I will check my old build scripts to see what I built against

**Reply by suesauer** [reply]
Maya usd : git checkout 00d651a8af3af752160531a0841579d1bc0f41ea #v.0.1.0
PXR usd git checkout 87e0dc5322be2794230e8e53d45bedcbe2d52a86  #version 19.11

these worked for me in 2019 and 2020.1

**Reply by mreinhardmikros** [reply]
Thank you, Sue!
That's not really what I was hoping for :stuck_out_tongue_winking_eye: 
But at least I have an indication that this was working at some point.
Cheers,
Marcel

**Comment by kxl-adsk** [comment]
We are building and running tests for every pull-request with Maya 2019. I just tried it with the latest maya-usd and latest supported USD dev commit. But this doesn't help you much. 

What's the platform you are running MayaUsd on? (Linux, Windows, MacOS)

**Reply by mreinhardmikros** [reply]
Well, that could actually help me a lot!
At least it tells me that I should be able to make things work somehow.
When you say you tried, does that mean that you were able to use the "Stage from file" command? Is this also included in your tests using an interactive session?
Last but not least: Since you are building for Maya 2019 yourself, you must have a recipe of how to build the last USD dev while still complying to the dependency versions imposed by Maya. Can you share any insights on this?
I am on CentOs 7 here.

Thanks for your help!

**Comment by kxl-adsk** [comment]
> When you say you tried, does that mean that you were able to use the "Stage from file" command?

Of course, proxy shape visualization with VP2RenderDelegate is supported in Maya 2019

> Since you are building for Maya 2019 yourself, you must have a recipe of how to build the last USD dev while still complying to the dependency versions imposed by Maya. Can you share any insights on this?

We do have a conversation in the beta forum on this. How about you share here your final solution, once we conclude on it? This way we avoid creating a situation where people without access to the beta forum can't fully benefit from this conversation.

Cheers,
KxL
